pkg/provider/data_sync.go

```diff
@@ -115,6 +121,12 @@ func dataSyncRead(ctx context.Context, d *schema.ResourceData, m interface{}) di
 	opt.Branch = d.Get("branch").(string)
 	opt.TargetPath = d.Get("target_path").(string)
 	opt.GitImplementation = d.Get("git_implementation").(string)
+	opt.Secret = d.Get("secret").(string)
+	// See #104: fallback to resource name
+	if opt.Secret == "" {
+		opt.Secret = opt.Name
+		d.Set("secret", opt.Name)
+	}
```

Comment:
I understand that there is a will to keep things backwards compatible, but what are the effects on end users when the secret is replaced as they upgrade from one version of the provider to another?

Reply:
For me, personally, backwards-compatibility is important - that's why i proposed the change as it is. If a user upgrades to 0.0.13 the secretRef is wrong users can't sync their repositories: There's no Ops in GitOps happening.
Right now there's nothing that a user can do about it other than creating a secret with the specific name `flux-system`. 

```tf
# example from the readme
resource "kubernetes_secret" "main" {
  metadata {
    name      = data.flux_sync.main.name # <-- this won't work any more.
  }

  data = {
    username = "git"
    password = "1234"
  }
}
```


We can certainly live with a backwards-incompatible change and do the change in our wrapper module. However i sense that we're not the only user of this tf provider.

Certainly, this is a very young project and there's AFAIK no compatibility guarantee. Let me know how you prioritize backwards-compat vs. pretty&simple code, i'll do the changes accordingly. The essential part is to make the `secret` name configurable.

Reply:
I would rather not implement this special case, as it will probably introduce problems in the future when changes are made again and this special case is removed.

So this will be a breaking change to people who set the name in `data_sync` to something other than the default value and expect the secret to follow along with the same name. I do not know how other people are using the provider but is the majority changing this value? Considering the fact that this issue made its way into 0.13 sadly the harm is to a certain degree already done. I totally agree with you that getting secret ref in should be done ASAP, I will create a release as soon as we merge this. 

After thinking about this during the weekend I think we should stick with using the default values set by default options in the fluxv2 repo. If you can make those changes I will merge this PR.

Reply:
sure, that works!